File: code/Diana0.01/Web/Templates/BootstrapTemplate/js/action.js
$(function () {
    gridList();
})
function gridList() {
    var $gridList = $("#gridList");
    $gridList.dataGrid({
        url: "../index.ashx?ctrl=action&action=showAll",
        height: $(window).height() - 126,
        //pager: 'pager',
        //rowList: [10, 20, 30], //可调整每页显示的记录数   
        //rowNum: 40,
        pager: "#gridPager",
        sortname: 'id asc',
        viewrecords: true,
        colModel: [
            { label: "主键", name: "id", hidden: true, key: true },
            { label: '动作名称', name: 'actionname', width: 150, align: 'left' },
            { label: '动作编号', name: 'actioncode', width: 150, align: 'left' },
            { label: '动作所属', name: 'actionowner', width: 150, align: 'left', },
            { label: '动作类型', name: 'actiontype', width: 150, align: 'left', },
            { label: '动作地址', name: 'actionurl', width: 150, align: 'left' },
            { label: '动作参数', name: 'actionparam', width: 150, align: 'left' },
            { label: '控制器名称', name: 'controlername', width: 150, align: 'left' }
        ]
    });
    $("#btn_search").click(function () {
        $gridList.jqGrid('setGridParam', {
            postData: { keyword: $("#txt_keyword").val() },
        }).trigger('reloadGrid');
    });
}
function btn_add() {
    $.modalOpen({
        id: "Form",
        title: "新增动作",
        url: "../index.ashx?ctrl=action&action=Form",
        width: "700px",
        height: "510px",
        callBack: function (iframeId) {
            top.frames[iframeId].submitForm();
        }
    });
}
function btn_edit() {
    var keyValue = $("#gridList").jqGridRowValue().id;
    $.modalOpen({
        id: "Form",
        title: "修改动作",
        url: "../index.ashx?ctrl=action&action=Form&keyValue=" + keyValue,
        width: "700px",
        height: "510px",
        callBack: function (iframeId) {
            top.frames[iframeId].submitForm();
        }
    });
}
function btn_delete() {
    $.deleteForm({
        url: "../index.ashx?ctrl=action&action=DeleteForm",
        param: { keyValue: $("#gridList").jqGridRowValue().id },
        success: function () {
            $.currentWindow().$("#gridList").resetSelection();
            $.currentWindow().$("#gridList").trigger("reloadGrid");
        }
    })
}

function btn_details() {
    var keyValue = $("#gridList").jqGridRowValue().id;
    $.modalOpen({
        id: "Details",
        title: "查看动作",
        url: "../index.ashx?ctrl=action&action=Details&keyValue=" + keyValue,
        width: "700px",
        height: "510px",
        btn: null,
    });
}
function btn_CopyAndPaste() {
    $.deleteForm({
        url: "../index.ashx?ctrl=action&action=copyAndPasteForm",
        param: { keyValue: $("#gridList").jqGridRowValue().id },
        success: function () {
            $.currentWindow().$("#gridList").trigger("reloadGrid");
        }
    })
}

File: code/Diana0.01/model/Model/action.cs
﻿using System;
using System.Linq;
using System.Text;
/// <summary>
///项目名称：Diana轻量级开发框架
///版本：0.0.1版
///开发团队成员：胡凯雨，张梦丽，艾美珍，易传佳，陈祚送，康文洋，张婷婷，王露，周庆，夏萍萍，陈兰兰
///模块和代码页功能描述：动作实体类
///最后修改时间：2018/1/26
/// </summary>
namespace Diana.model
{
    [Serializable]
    public class action
    {
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int id { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string actionname { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? actiontype { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string actionowner { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int actioncode { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string actionurl { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string actionparam { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string controlername { get; set; }
        /// <summary>
        /// 
        /// </summary> 
        public string actiondescription { get; set; }
    }
}
